clang/lib/StaticAnalyzer/Checkers/StreamChecker.cpp

```diff
@@ -880,6 +883,24 @@ void StreamChecker::preReadWrite(const FnDescription *Desc,
   }
 }
 
+void StreamChecker::preWrite(const FnDescription *Desc, const CallEvent &Call,
```

Comment:
Why 　we need a separated `preWrite` ? The original `preReadWrite` also works.

Reply:
The `CallDescriptionMap` was uncomfortable to handle because too many `std::bind` functions.

Reply:
Let me share my experience with `std::bind` and with this checker.
Unfortunately, we also have/had a couple (<10) conflicting patches to this checker, and I had to rebase 16 times to clang-18. This is fine, but when I this many `std::binds`, and sometimes changes where the railing "bool" flag meaning was changed, I really had to be on my toes when resolving syntactic & semantic conflicts.
I feel the `std::bind` usage got out of hand here, and I now regret not pushing back.

The bottom line is, that I think having dedicated handlers dispatching to common implementations, or just copy-pasting their implementation while specializing it leads to cleaner code than using `std::bind` and directly binding to the generic implementation.

Reply:
I agree that `std::bind` is hard to follow, and dedicated handlers dispatching to common implementations is a better model.

The "Copy-pasting their implementation while specializing it" could be even better if the common parts are lifted out into helper functions; but I strongly suggest that we should avoid code duplication.